src/R/Components/Impl/ConnectionManager/Implementation/ViewModel/ConnectionViewModel.cs

```diff
@@ -183,15 +183,29 @@ internal sealed class ConnectionViewModel : BindableBase, IConnectionViewModel {
             IsRemote = !(uri?.IsFile ?? true);
         }
 
-       public string GetCompletePath() {
+        public void PathEditCompleted() {
+            // After the user is done editing the path, we fix it up automatically
+            Path = GetCompletePath();
+
+            // Conveniently populate the name, if the user directly edited the path
+            // without specifying a name.
+            if (string.IsNullOrWhiteSpace(Name)) {
```

Comment:
One catch here is that the first time they leave focus, we'll auto-generate the name, but if they later update the URI the name will stay as is. I think this should rather check if the name is empty or looks like it was auto-generated (from the URI before the edit), and update it in the latter case also.